src/main/java/com/sarality/datasource/AggregateQueryDataSource.java

```diff
@@ -0,0 +1,48 @@
+package com.sarality.datasource;
+
+import com.sarality.db.SQLiteTable;
+import com.sarality.db.TableRegistry;
+import com.sarality.db.cursor.CursorDataExtractor;
+import com.sarality.db.cursor.JoinQueryCursorDataExtractor;
+import com.sarality.db.query.SimpleJoinQueryBuilder;
+import com.sarality.db.query.SimpleQueryBuilder;
+
+import java.util.List;
+
+/**
+ * Reads data from a table using an aggregate query and a cursor data extractor
+ *
+ * @author satya@ (Satya)
+ */
+public class AggregateQueryDataSource<K, T> implements DataSource<List<K>> {
```

Comment:
Is this really an aggregate data source or just a TableViewDataSource or TableQueryDataSource? Basically you have a table and you are overlaying a view on it so that you get an object of K rather than an object of type T from it.